app/controllers/api_controller/chargeback.rb

```diff
@@ -0,0 +1,25 @@
+class ApiController
+  module Chargeback
+    # Request
+    # {
+    #   "action":"assign",
+    #
+    #   "data": [
+    #     {
+    #         "object_type": "tenant",
+    #         "object_id": 1
+    #     }
+    #   ]
+    # }
+    def assign_resource(_type, id, data = nil)
```

Comment:
I think there is some metaprogramming going on that targets assigning of resources - for this it would be something like `chargebacks_assign_resource`. I'm not sure how this works, but if it does it would benefit from either hooking into the above if possible, or if not having a more specific name (all these methods get mixed into the ApiController, making a collision quite likely with such a generic name).